echo-scheduler/src/test/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/MissedPipelineTriggerCompensationJobSpec.groovy

```diff
@@ -0,0 +1,195 @@
+        new OrcaService.PipelineResponse(pipelineConfigId: '4', startTime: getDateOffset(30).time)
+      ])
+    }
+    1 * pipelineInitiator.call((Pipeline) pipelines[0])
+    0 * _
+  }
+
+  def 'should no-op with no missed executions'() {
+    given:
+    def pipelines = [
+      pipelineBuilder('1').disabled(false).triggers([
+        new Trigger.TriggerBuilder().id('1').type(Trigger.Type.CRON.toString()).cronExpression('* 0/30 * * * ? *').enabled(true).build()
+      ]).build(),
+      pipelineBuilder('2').disabled(false).triggers([
+        new Trigger.TriggerBuilder().id('2').type(Trigger.Type.CRON.toString()).cronExpression('* 0/30 * * * ? *').enabled(true).build()
+      ]).build()
+    ]
+
+    and:
+    def dateContext = new MissedPipelineTriggerCompensationJob.DateContext(
+      timeZone: TimeZone.getTimeZone('America/Los_Angeles'),
+      triggerWindowFloor: getDateOffset(0),
+      now: getDateOffset(0)
+    )
+    def compensationJob = new MissedPipelineTriggerCompensationJob(scheduler, pipelineCache, orcaService, pipelineInitiator, counterService, 30000, 'America/Los_Angeles', dateContext)
+
+    when:
+    compensationJob.triggerMissedExecutions(pipelines)
+
+    then:
+    1 * orcaService.getLatestPipelineExecutions(_) >> {
+      rx.Observable.just([
+        new OrcaService.PipelineResponse(pipelineConfigId: '1', startTime: getDateOffset(0).time)
+      ])
+    }
+    0 * pipelineInitiator.call(_)
+    0 * _
+  }
+
+  def 'should use only enabled cron triggers'() {
+    given:
+    def pipelines = [
+      pipelineBuilder('1').triggers([
+        new Trigger.TriggerBuilder().id('1').type(Trigger.Type.CRON.toString()).enabled(true).build(),
+        new Trigger.TriggerBuilder().id('2').type(Trigger.Type.CRON.toString()).enabled(false).build()
+      ]).build(),
+      pipelineBuilder('2').build(),
+      pipelineBuilder('3').triggers([
+        new Trigger.TriggerBuilder().id('3').enabled(true).build(),
+        new Trigger.TriggerBuilder().id('4').enabled(false).build(),
+        new Trigger.TriggerBuilder().id('5').type(Trigger.Type.CRON.toString()).enabled(true).build()
+      ]).build()
+    ]
+
+    when:
+    def result = MissedPipelineTriggerCompensationJob.getEnabledCronTriggers(pipelines)
+
+    then:
+    result.collect { it.id } == ['1', '5']
+  }
+
+  @Unroll
+  def 'should evaluate missed executions only in window'() {
+    given:
+    def clock = Clock.systemDefaultZone()
+
+    def expr = new CronExpression(cronExpression)
+    expr.timeZone = TimeZone.getTimeZone(clock.zone)
+
+    def lastExecution = getDateOffset(lastExecutionMinutes)
+    def windowFloor = getDateOffset(windowFloorMinutes)
+    def now = getDateOffset(nowMinutes)
+
+    when:
+    def result = MissedPipelineTriggerCompensationJob.missedExecution(expr, lastExecution, windowFloor, now)
+
+    then:
+    result == missedExecution
+
+    where:
+    cronExpression     | lastExecutionMinutes | windowFloorMinutes | nowMinutes | missedExecution
```

Comment:
minor convention but I'd suggest using `||` as a delimiter between inputs and expected results ...

```
windowFloorMinutes | nowMinutes || missedExecution
```